src/main/java/shef/servlets/UserServlet.java

```diff
@@ -99,29 +102,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String location = request.getParameter("location-input");
     String profilePictureUrl = getUploadedFileUrl(request, "profile-pic-upload");
     String bio = request.getParameter("bio-input");
+    String[] parameters = { email, username, location, profilePictureUrl, bio };
```

Comment:
I tried with getParameterMap, but ultimately abandoned the method because 1) it maps a name to an array of values instead of a single value, and 2) three of the parameters don't come from the the request anyways (the id, the email, and the pictureUploadUrl)

Reply:
Why did you choose to use two arrays instead of a map?

Reply:
I see the issue now, I will change the implementation to a map.